pkg/admission/webhook/generic_webhook.go

```diff
@@ -96,12 +96,13 @@ func (p *WebhookDispatcher) getAPIBindingWorkspace(attr admission.Attributes, cl
 	for _, obj := range objs {
 		apiBinding := obj.(*apisv1alpha1.APIBinding)
 		for _, br := range apiBinding.Status.BoundResources {
-			if apiBinding.Status.BoundAPIExport == nil || apiBinding.Status.BoundAPIExport.Workspace == nil {
+			if apiBinding.Status.BoundAPIExport != nil && apiBinding.Status.BoundAPIExport.Workspace == nil {
 				// this will never happen today. But as soon as we add other reference types (like exports), this log output will remind out of necessary work here.
 				klog.Errorf("APIBinding %s has no referenced workspace", clusterName, apiBinding.Name)
 				continue
 			}
-			if br.Group == attr.GetResource().Group && br.Resource == attr.GetResource().Resource {
+			if apiBinding.Status.BoundAPIExport != nil && apiBinding.Status.BoundAPIExport.Workspace != nil &&
+				br.Group == attr.GetResource().Group && br.Resource == attr.GetResource().Resource {
```

Comment:
💣 this potentially means that bound resources will miss admission. That's not good. We have to think and clean up our binding state logic everywhere.